h/tasks/job_queue.py

```diff
@@ -0,0 +1,6 @@
+from h.celery import celery
+
+
+@celery.task
+def sync_annotations():
+    celery.request.find_service(name="job_queue").sync_annotations()
```

Comment:
It depends what you see the job queue as.

I see it as a general-purpose job queue that has different properties from the Celery task queue (it's transactional with the DB etc). It just so happens to only be used for sync-annotation jobs currently. And may only ever be used for those jobs. But it could be used in the future for some other type of job that has nothing to do with annotations or the search index. So I see the job queue as its own separate thing, not part of the `search_index` service.

The Celery task for consuming from the job queue though (`JobQueue.sync_annotations()`) could possibly call a `search_index` service method to do the actual syncing of a batch of annotations from Postgres to Elastic? Since syncing annotations is a search index thing. The split of responsibilities could be awkward though as reading a batch of jobs from the job queue and possibly deleting those jobs from the queue seems like a job queue thing, whereas reading annotation IDs from Postgres and Elastic and possibly syncing them into Elastic is maybe a search index thing. So it seems that the algorithm that's currently in one method would have to be split up into two places

Reply:
I think it's pretty clear right now it's job is 1:1 with indexing. In the event that we ever did something else with it, I think that would be the time to think about disentangling it.